Homework L08

dacian-cioc opened the pull request:
Inheritance & Encapsulation homework

dacian-cioc committed:
Homework L08

Inheritance & Encapsulation homework

dacian-cioc committed:
homework 8 - small changes

KaloyanTsvetkov commented on src/L08/homework/Lector.java:
```diff
@@ -0,0 +1,55 @@
+
+    // Method to calculate overtime payment for experienced lecturers
+    public double calculateOvertime(double hours) {
+        if (getAge() > 30) {
```
number 30 could be a constant field and will bring more information that just using the number itself:
EXPERIENCED_LECTOR_MIN_AGE

KaloyanTsvetkov commented on src/L08/homework/Lector.java:
```diff
@@ -0,0 +1,55 @@
+    public double calculateOvertime(double hours) {
+        if (getAge() > 30) {
+            // Experienced lecturers receive overtime payment
+            return (dailySalary / 8) * hours * 2; // Assuming 8 hours per working day
```
number 8 could be a constant field and will bring more information that just using the number itself:
WORKING_DAY_IN_HOURS

KaloyanTsvetkov commented on src/L08/homework/Demo.java:
```diff
@@ -0,0 +1,44 @@
+            } else if (person instanceof Lector) {
+                ((Lector) person).showLectorInfo();
+                // Calculate and display overtime for lectors
+                if (person.getAge() > 30) {
```
This if statement is not needed. You have such validation in the 'calculateOvertime' method. It will just return 0 for overtime.

dacian-cioc committed:
Updating based on comments

adding EXPERIENCED_LECTOR_MIN_AGE, WORKING_DAY_IN_HOURS in lector class and removing redundant if statement in demo class
